app/helpers/episodes_helper.rb

```diff
@@ -16,21 +16,26 @@ def episode_explicit_options
   end
 
   def episode_apple_status(episode)
-    if episode.apple_status.nil?
+    apple_episode = episode.apple_episode
+    if !apple_episode
       "not_found"
-    elsif episode.apple_status.delivered?
+    elsif apple_episode.needs_delivery?
+      "incomplete"
+    elsif apple_episode.waiting_for_asset_state?
+      "processing"
+    elsif apple_episode.audio_asset_state_error?
+      "error"
+    elsif apple_episode.synced_with_apple?
```

Comment:
Are these API requests?  I don't see an `apple_episodes` table in the db, so not entirely clear where these statuses are stored.

Reply:
Much of this status comes from the Apple API, but uses the latest responses which are stored.

Reply:
Yep that's right. Almost all the Apple models mix in a `Apple::ApiResponse` + `has_one :apple_sync_log`.  The "SyncLog" is the last recorded remote state (usually from polling or updating the remote resource).

Apple returns json:api style responses from their endpoints. We record the last response for a remote resource in the local resource's "sync log".  The ApiResponse mixin wraps the SyncLog, and [gives some simple helpers](https://github.com/PRX/feeder.prx.org/blob/main/app/models/apple/api_response.rb#L77-L112) to unwrap json:api data.

Since the Apple::Episode is not a db table, [that class references the base ::Episode model's SyncLog by overriding the attr reader](https://github.com/PRX/feeder.prx.org/blob/main/app/models/apple/episode.rb#L219-L221).

Lastly there is the media status. [This has it's own table that records attributes of the media (version, filename, CDN source, etc](https://github.com/PRX/feeder.prx.org/blob/main/app/models/concerns/apple_delivery.rb#L33-L35).

So the result here is that `synced_with_apple?` here wraps delivery/podcast container/apple_episode state as well as uploaded media state. It uses a variety of helper methods to look down into the last stored apple api responses as well as the records of what media has been uploaded so far.